src/typecheck/arrays.ml

```diff
@@ -1,6 +1,42 @@
 
 module Id = Dolmen.Std.Id
 
+(* Ae arrays *)
+(* ************************************************************************ *)
+
+module Ae = struct
+
+  module Tff
+      (Type : Tff_intf.S)
+      (Ty : Dolmen.Intf.Ty.Ae_Array with type t := Type.Ty.t)
+      (T : Dolmen.Intf.Term.Ae_Array with type t := Type.T.t) = struct
+
+    type _ Type.err +=
+      | Bad_farray_arity : Dolmen.Std.Term.t Type.err
+
+    let parse env s =
+      match s with
+      | Type.Id { name = Simple "farray"; ns = Term } ->
+        `Ty (
+          fun ast args ->
+            match args with
+            | [_] ->
+              let int_ty = Dolmen_std.Term.builtin Dolmen_std.Term.Int () in
+              Base.ty_app2 (module Type) env s Ty.array ast (int_ty :: args)
```

Comment:
Since you already check the arity explicitly, you don't need the `Base.ty_app` wrapper here, and can directly return the result of applying `Ty.array` to the correct args. (same for the second case).

Reply:
Done.